packages/create-cloudflare/src/frameworks/next/index.ts

```diff
@@ -53,26 +55,38 @@ const getApiTemplate = (
 const configure = async (ctx: PagesGeneratorContext) => {
 	const projectName = ctx.project.name;
 
-	// Add a compatible function handler example
-	const path = probePaths(
-		[
-			`${projectName}/pages/api`,
-			`${projectName}/src/pages/api`,
-			`${projectName}/src/app/api`,
-			`${projectName}/app/api`,
-			`${projectName}/src/app`,
-			`${projectName}/app`,
-		],
-		"Could not find the `/api` or `/app` directory"
-	);
+	const path = probePaths([
+		`${projectName}/pages/api`,
+		`${projectName}/src/pages/api`,
+		`${projectName}/src/app/api`,
+		`${projectName}/app/api`,
+		`${projectName}/src/app`,
+		`${projectName}/app`,
+	]);
+
+	if (!path) {
+		return crash("Could not find the `/api` or `/app` directory");
```

Comment:
I am `return`ing here so that below `path` gets correctly inferred as `string`, if I were to simply call `crash` then `path` would be inferred as `string|null` 😓 

(alternatively I can use a non-null assertion if that's preferred)

Reply:
(PS: I am not sure why typescript is not detecting that `crash` stops the flow here 😓)

Reply:
we already seem to be doing this in some other places as well (https://github.com/cloudflare/workers-sdk/blob/037de5ec77efc8261860c6d625bc90cd1f2fdd41/packages/create-cloudflare/src/frameworks/index.ts#L43) so hopefully it's fine 🙂

Reply:
Yeah that's fine. Without it TS thinks the function might return `undefined`.

Reply:
In general, we should avoid `process.exit()` in code that is not directly in the `main()` function. It makes it nigh on impossible to write tests for.
Instead we should throw errors that bubble up.

Reply:
> Yeah that's fine. Without it TS thinks the function might return undefined.

But the return type is `never`, isn't that supposed to signal to typescript that the function doesn't return/breaks the current function's flow? 🤔

Reply:
Hmm.. Interesting. it seems that declaring `crash` as a `const` variable to a function that returns never is different to declaring it as a `function`.

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGpwDbIN7Jge3wAoBKALmQGcwpQBzZAXwCgATCBDOKFGAVxAIwwfCGQQAHsDCkKICADdoAbmbN+g4aOQIocSgAtZyeUqg5mASEnTSqlswSjqOvYYBMyALzJS3gHwW1lIyJPZqGkIiYgC2cKAAjEQKmBToGAA+AuwwoBCsJBbIyMAwvilYXlXI2RC58gVFxa76RiTMxSzFPGB8UGIVAHQExGHMDpFasfEg7smpaJhZIDl5jdgdJWXzldW19fmFG80tHqSbXcg9fQOYw4R242pAA

Reply:
Although if the function has an explicit typing then it works:

```ts
export const crash: (msg?: string) => never = (msg) => {
	if (msg) {
		process.stderr.write(red(msg));
		process.stderr.write("\n");
	}
	exit(1);
};
```

Reply:
ah wow, nice finding! 😃 

I've updated the code so that we then use an explicit type for the `crash` function so that we can avoid these awkward `return`s, please let me know if you're ok with it 🙂